src/MatrixSecondary.Java

```diff
@@ -0,0 +1,129 @@
+     *
+     * @param mtrx the matrix to add
+     * @requires mtrx.rows() == rows() and mtrx.columns() == columns()
+     * @updates this
+     * @ensures this.getEntry(i, j) == (this.getEntry(i, j)) + mtrx.getEntry(i, j) for all is and js.
+     */
+    void elementAdd(MatrixKernel mtrx){
+        assert this.rows() == mtrx.rows;
+        
+        int rows = mtrx.rows();
+        int columns = mtrx.columns();
+
+        for (int r = 0; r < rows; r++) {
+            for (int c = 0; c < columns; c++) {
+                double newEntry = this.getEntry(r, c) + mtrx.getEntry(r, c);
+                this.changeEntry(r, c, newEntry);
+            }
+        }
+    }
+
+    /**
+     * Multiplies this matrix by another matrix element-wise.
+     *
+     * @param mtrx the matrix to multiply
+     * @requires mtrx.rows() == rows() and mtrx.columns() == columns()
+     * @updates this
+     * @ensures this.getEntry(i, j) == (this.getEntry(i, j)) * mtrx.getEntry(i, j) for all is and js.
+     */
+    void elementMultiply(Matrix mtrx){
+        assert this.rows() == mtrx.rows;
+
+        int rows = mtrx.rows();
+        int columns = mtrx.columns();
+
+        for (int r = 0; r < rows; r++) {
+            for (int c = 0; c < columns; c++) {
+                double newEntry = this.getEntry(r, c) * mtrx.getEntry(r, c);
+                this.changeEntry(r, c, newEntry);
+            }
+        }
+    }
+
+    /**
+     * Computes the matrix multiplication of this matrix and another matrix.
+     *
+     * @param mtrx the matrix to multiply with
+     * @return a new matrix representing the product
+     * @requires columns() == mtrx.rows()
+     * @ensures this.rows() == rows() and result.columns() == mtrx.columns(), a matrix following matrix mathmatical matrix multiplication.
+     */
+    MatrixEnhanced matrixMultiply(Matrix mtrx){
+        assert this.columns() == mtrx.rows();
+
+        int thisRows = this.rows();
+        int mtrxColumns = mtrx.columns();
+        int commonNum = this.columns();
+
+        Matrix result = new Matrix(thisRows, mtrxColumns);
+
+        for (int i = 0; i < thisRows; i++) {
+            for (int j = 0; j < mtrxColumns; j++) {
+                double sum = 0.0;
+                for (int k = 0; k < commonNum; k++) {
+                    sum += this.getEntry(i, k) * mtrx.getEntry(k, j);
+                }
+                result.changeEntry(i, j, sum);
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Computes the dot product of this matrix and another matrix by summing all
+     * element-wise additions of their entries.
+     *
+     * @param mtrx the matrix to add element-wise
+     * @return the sum of all entries after addition
+     * @requires mtrx.rows() == rows() and mtrx.columns() == columns()
+     * @ensures dotProduct is the sum of (this.getEntry(i, j) + mtrx.getEntry(i, j)) for all is and js
+     */
+    double dotProduct(Matrix mtrx){
```

Comment:
Docs are needed here since they're inherited. You can however label these as `@Override`